### ISSUE TITLE
Detekt - Enable Type Resolution (detektMain/Test)

### ISSUE THREAD

**Issue by ParaskP7** [wordpress-mobile/WordPress-Android#17011]
Parent #16892

This issue is about enabling type resolution for Detekt via `detektMain/Test` tasks.

**TL;DR**: We might want to start using `detektMain/Test` in favor of `detekt` task as it is nowadays the recommend way to configure Detekt (see [Using Type Resolution](https://detekt.dev/docs/gettingstarted/type-resolution) and more specifically [here](https://detekt.dev/docs/gettingstarted/type-resolution/#enabling-on-a-jvm-project) and [here](https://detekt.dev/docs/gettingstarted/type-resolution/#enabling-on-an-android-project)). The plain `detekt` task is not running with type resolution enabled, while the `detektMain/Test` tasks are (for more info watch this nice Detekt related talk [here](https://www.droidcon.com/2021/11/17/detekt-state-of-the-union-2/), go to 30:30, up until 34:00).

**Comment by ParaskP7** [comment]
Closing this because its associated Detekt apps infra request (pdnsEh-qb-p2) has been marked completed (pdnsEh-qb-p2#comment-1072) for some time already and it is unlikely that more work will be done on that front anytime soon.
